src/ecwrapper.cpp

```diff
@@ -178,17 +178,11 @@ bool CECKey::Recover(const uint256 &hash, const unsigned char *p64, int rec)
 {
     if (rec<0 || rec>=3)
         return false;
-    ECDSA_SIG *sig = nullptr;
-    BIGNUM *sig_r = nullptr, *sig_s = nullptr;
-    if (!(sig = ECDSA_SIG_new()) ||
-        !(sig_r = BN_bin2bn(&p64[0],  32, nullptr)) ||
-        !(sig_s = BN_bin2bn(&p64[32], 32, nullptr)) ||
-        !ECDSA_SIG_set0(sig, sig_r, sig_s)) {
-        ECDSA_SIG_free(sig);
-        BN_free(sig_r);
-        BN_free(sig_s);
-        return false;
-    }
+    ECDSA_SIG *sig = ECDSA_SIG_new();
+    BIGNUM *sig_r = BN_bin2bn(&p64[0],  32, nullptr);
+    BIGNUM *sig_s = BN_bin2bn(&p64[32], 32, nullptr);
+    assert(sig && sig_r && sig_s);
+    assert(ECDSA_SIG_set0(sig, sig_r, sig_s));
```

Comment:
Note that this relies on ``assert`` not being compiled out. In fact we require asserts not to be compiled out in general, but I would still prefer it to be:
```
bool ret = ECDSA_SIG_set0(sig, sig_r, sig_s);
assert(ret);
```